uniconnect-epfl/uniconnect
# Final end to end test and fixed

# What I did
I solved the an issue showing user events in external profile instead of the contact events

# How I did it
I modularised the Profile events allowing to pass in argument the correct userId. The component can now be re-used anywhere.
I added the end-to-end test for the M4 showing interactions with the app, including registration, login and walkthrough of the app

# How to verify it
You can now check externals profiles the events should render correctly for your friends

You can run the maestro tests and observe the results in you emulator, like previous PR. Put the apk in your emulator.
You first have to install maestro with `curl -Ls "https://get.maestro.mobile.dev" | bash` and then run `maestro test <relative_path>`
![image](https://github.com/uniconnect-epfl/uniconnect/assets/108724823/d5a51c77-e19d-44b9-ba4c-3e9f2c5d73f2)

https://github.com/uniconnect-epfl/uniconnect/assets/108724823/3327ac67-787c-4244-b5f9-191d379e7703




if the end to end test fail on another's emulator it could be due to the user being authenticated with google and pop up screen could interfer.

# Pre-merge checklist
The changes I have introduced:
